cider-dap/src/main.rs

```diff
@@ -164,18 +182,114 @@ fn run_server<R: Read, W: Write>(
-            // Here, can add a match pattern for a disconnect or exit command
-            // to break out of the loop and close the server.
-            // Command::Disconnect(_) => break,
-            // ...
+            // Disconnect the server AND exit the debugger
+            Command::Disconnect(_) => {
+                let rsp = req.success(ResponseBody::Disconnect);
+                server.send_event(Event::Exited(ExitedEventBody {
+                    exit_code: 0,
+                }))?;
+                server.respond(rsp)?;
+
+                //Exit
+                eprintln!("exited debugger");
+                return Ok(());
+            }
+            // Send StackTrace, may be useful to make it more robust in the future
+            Command::StackTrace(_args) => {
+                let rsp =
+                    req.success(ResponseBody::StackTrace(StackTraceResponse {
+                        stack_frames: vec![],
+                        total_frames: None,
+                    }));
+                server.respond(rsp)?;
+            }
+            // Continue the debugger
+            Command::Continue(_args) => {
+                let rsp =
+                    req.success(ResponseBody::Continue(ContinueResponse {
+                        all_threads_continued: None,
+                    }));
+                server.respond(rsp)?;
+            }
+            // Send a Stopped event with reason Pause
+            Command::Pause(args) => {
+                // Get ID before rsp takes ownership
+                let thread_id = args.thread_id;
+                let rsp = req.success(ResponseBody::Pause);
+                // Send response first
+                server.respond(rsp)?;
+                // Send event
+                server.send_event(Event::Stopped(StoppedEventBody {
+                    reason: types::StoppedEventReason::Pause,
+                    description: Some(String::from("Paused")),
+                    preserve_focus_hint: None,
+                    text: None,
+                    thread_id: Some(thread_id),
+                    all_threads_stopped: None,
+                    hit_breakpoint_ids: None,
+                }))?;
+            }
+            // Step over
+            Command::Next(args) => {
+                // Get ID before rsp takes ownership
+                let thread_id = args.thread_id;
+                let rsp = req.success(ResponseBody::Next);
+                // Send response first
+                server.respond(rsp)?;
+                // Send event
+                server.send_event(Event::Stopped(StoppedEventBody {
+                    reason: types::StoppedEventReason::Step,
+                    description: Some(String::from("Continue")),
+                    thread_id: Some(thread_id),
+                    preserve_focus_hint: None,
+                    text: None,
+                    all_threads_stopped: None,
+                    hit_breakpoint_ids: None,
+                }))?;
```

Comment:
Since you repeat this snipped a lot, probably better to create a constructor that takes the description string and thread_id as arguments and returns the stopped event